jwst/assign_wcs/nirspec.py

```diff
@@ -472,50 +487,62 @@ def get_open_msa_slits(msa_file, msa_metadata_id, slit_y_range=[-.55, .55]):
     # of the configuration file.
     try:
         msa_file = fits.open(msa_file)
-    except:
-        message = "Unable to open MSA FITS file (MSAMETFL) {0}".format(msa_file)
+    except FileNotFoundError:
+        message = "Missing MSA meta (MSAMETFL) file {}".format(msa_file)
         log.error(message)
-        raise MissingMSAFileError(message)
+        raise MSAFileError(message)
+    except OSError:
+        message = "Unable to read MSA FITS file (MSAMETFL) {0}".format(msa_file)
+        log.error(message)
+        raise MSAFileError(message)
+    except Exception:
+        message = "problem reading MSA metafile (MSAMETFL) {0}".format(msa_file)
+        log.error(message)
+        raise MSAFileError(message)
 
     # Get the configuration header from teh _msa.fits file.  The EXTNAME should be 'SHUTTER_INFO'
     msa_conf = msa_file[('SHUTTER_INFO', 1)]
     msa_source = msa_file[("SOURCE_INFO", 1)].data
 
     # First we are going to filter the msa_file data on the msa_metadata_id
     # as that is all we are interested in for this function.
```

Comment:
Update the comment here so that it's clear we're now filtering on both msa_metadata_id AND dither_point_index